packages/dds/tree/src/core/schema-stored/storedSchemaRepository.ts

```diff
@@ -48,6 +71,8 @@ export class InMemoryStoredSchemaRepository<TPolicy extends SchemaPolicy = Schem
 {
     readonly computationName: string = "StoredSchemaRepository";
     protected readonly data: MutableSchemaData;
+    public readonly emitter = EventEmitter.create<SchemaEvents>();
```

Comment:
Have you considered having `InMemoryStoredSchemaRepository` implement `IEventEmitter` directly, and then making this `emitter` field private? It gives subscribers a less verbose API and is more consistent with current patterns in the FluidFramework (things that expose events have an `on` method directly on them, rather than having a field for accessing the events). It does require a small amount of boilerplate to implement the `on` method and delegate to the emitter, but its trivial and only 3 lines. I think it's a worthwhile tradeoff. This applies to the other places where you've added event emitters too, with perhaps the exception of `IForestSubscription`; in my mind it's a more abstract "connection" to a concrete object rather than being a concrete object, and therefore maybe it shouldn't have subscriptions directly on it, but I admit that's totally subjective :)